### PR TITLE
Added pytest_args option to `additional_repos`

### PR DESCRIPTION
**Pull request by bmbouter**
[noissue]

### CODE REVIEW

**Comment by bmbouter** [comment]
I don't love this approach because it's very specific to the pulpcore use case. Maybe we should be auto-running the tests from the `additional_repos` data, or make a new section in the plugin_template?

**Review by fao89 — APPROVED** [review]

**Comment by dkliban** [comment]
This approach is fine. It's the only place where we run additional plugin's tests.

**Comment by fao89** [comment]
> I don't love this approach because it's very specific to the pulpcore use case. Maybe we should be auto-running the tests from the `additional_repos` data, or make a new section in the plugin_template?

you can have it on `func_test_script.sh`
https://github.com/pulp/pulp_file/blob/main/.github/workflows/scripts/func_test_script.sh

**Review by dkliban — APPROVED** [review]

**Comment by bmbouter** [comment]
Using `additional_repos` gets complicated when you have pulp_ansible using it like this https://github.com/pulp/pulp_ansible/blob/main/template_config.yml#L6-L10 So if we want to make this more generalized I would make a new section in template_config like `additional_functional_tests`?

**Comment by bmbouter** [comment]
I could have it be in `func_test_script.sh` but I'm wondering, does pulp_file need to run the certguard tests, and does pulpcore also need to run the pulp-certguard tests? If so it would be better to make this feature more general. I'm not sure. Maybe we should just use this for now and do the bigger generalization later.

**Comment by bmbouter** [comment]
Let's not merge this, I'm working on the more generalized version.

**Comment by bmbouter** [comment]
Check this out https://github.com/pulp/pulpcore/pull/2432

File: README.md
Comment (mdellweg) [inline README.md:57]: I'd expect a list of plugins from which we also run the functional tests.

File: templates/github/.github/workflows/scripts/script.sh.j2
Comment (mdellweg) [inline templates/github/.github/workflows/scripts/script.sh.j2:202]: Can we combine these in a single pytest invocation? Maybe add an option to stop trying after 20 failed tests.

File: README.md
Comment (mdellweg) [reply README.md:57]: Ah, I misread. The context in the diff was not enough without expanding.

Comment (mdellweg) [reply README.md:57]: But also to me that sounds like a feature switch:
`--pyargs              try to interpret all arguments as python packages.`

So technically we only add arguments to pytest, and the switch tells pytest to interpret _everything_ as python modules.

Comment (bmbouter) [reply README.md:57]: This is better I think because it doesn't rely on the assumption that all tests are in a canonical place. Also it's simpler because I don't have to build that string up in code, the data can just have it.

File: templates/github/.github/workflows/scripts/script.sh.j2
Comment (bmbouter) [reply templates/github/.github/workflows/scripts/script.sh.j2:202]: I think having them separate is kind of nice, that way if there is a failure you have big clear blocks showing you exactly where it is instead of one big run. The stopping after some time sounds nice but its not really the goal of this work.

File: README.md
Comment (mdellweg) [reply README.md:57]: I agree (see my second comment) i just don't really dig the name. Is it `test_module`?

File: templates/github/.github/workflows/scripts/script.sh.j2
Comment (mdellweg) [reply templates/github/.github/workflows/scripts/script.sh.j2:202]: The thing is that we stop processing after the first failed block. If we can combine test suites in one pytest instantiation, you would see all the failed tests you need to fix to make the CI green again.
Also we loose half of the "--parallel" benefit.
And now i realize the parallel tests suffer from the same restriction. They cannot be called in the same run with the non parallel ones.
I'm on the fence here.

File: README.md
Comment (mdellweg) [reply README.md:57]: What i like about this approach is that you could specify a subset of tests to run with pulpcore. But i'd rather have a `@pytest.mark.run_with_core` for this to add visibility.

Comment (bmbouter) [reply README.md:57]: I named it with `pyargs` because it's possible we may want to add a section that doesn't use `pyargs` perhaps? Like passing the string directly to pytest.

**Comment by bmbouter** [comment]
I renamed this to `pytest_args` and updated the README.

**Comment by bmbouter** [comment]
@mdellweg I'm going to merge this, but lmk if there is more that I should do to it. I suspect I'll be improving this machinery more over time.

**Pull request merged**
